File: market-rotation-tracker/src/config/fidelityFunds.ts
// Fidelity Select fund → parent ETF signal mapping.
// Used by the 401k tracker to translate rotation signals into fund recommendations.
// BrokerageLink accounts can also trade ETFs directly — ETF column is the signal source.
export interface FidelityFund {
    ticker: string;
    name: string;
    parentEtf: string;       // ETF we use as the signal proxy
    shortTermFee: boolean;   // true = 0.75% redemption fee if sold within 30 days
    minHoldDays: number;     // minimum hold to avoid short-term fee
}

export const FIDELITY_FUNDS: FidelityFund[] = [
    { ticker: 'FSPTX',  name: 'Fidelity Select Technology',            parentEtf: 'XLK',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FSELX',  name: 'Fidelity Select Semiconductors',        parentEtf: 'SMH',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FDCPX',  name: 'Fidelity Select Tech Hardware',         parentEtf: 'XLK',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FIDSX',  name: 'Fidelity Select Financial Services',    parentEtf: 'XLF',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FSRBX',  name: 'Fidelity Select Banking',               parentEtf: 'XLF',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FSPHX',  name: 'Fidelity Select Health Care',           parentEtf: 'XLV',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FIDRX',  name: 'Fidelity Select Industrials',           parentEtf: 'XLI',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FSDAX',  name: 'Fidelity Select Defense & Aerospace',   parentEtf: 'XLI',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FSUTX',  name: 'Fidelity Select Utilities',             parentEtf: 'XLU',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FSENX',  name: 'Fidelity Select Energy',                parentEtf: 'XLE',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FSLEX',  name: 'Fidelity Environment & Alt Energy',     parentEtf: 'XLU',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FRESX',  name: 'Fidelity Real Estate Fund',             parentEtf: 'XLRE', shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FSDPX',  name: 'Fidelity Select Materials',             parentEtf: 'XLB',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FSCSX',  name: 'Fidelity Select IT Services',           parentEtf: 'XLK',  shortTermFee: true,  minHoldDays: 30 },
    { ticker: 'FBGRX',  name: 'Fidelity Blue Chip Growth',             parentEtf: 'QQQ',  shortTermFee: false, minHoldDays: 0  },
];

// Look up fund by ticker
export function getFundByTicker(ticker: string): FidelityFund | undefined {
    return FIDELITY_FUNDS.find(f => f.ticker === ticker);
}

// Get all funds that map to a given ETF signal
export function getFundsByEtf(etf: string): FidelityFund[] {
    return FIDELITY_FUNDS.filter(f => f.parentEtf === etf);
}

File: market-rotation-tracker-kotlin/src/main/kotlin/com/rotationtracker/config/FidelityFunds.kt
package com.rotationtracker.config

data class FidelityFund(
    val ticker: String,
    val name: String,
    val parentEtf: String,
    val shortTermFee: Boolean,
    val minHoldDays: Int,
)

val FIDELITY_FUNDS: List<FidelityFund> = listOf(
    FidelityFund("FSPTX",  "Fidelity Select Technology",           "XLK",  true,  30),
    FidelityFund("FSELX",  "Fidelity Select Semiconductors",       "SMH",  true,  30),
    FidelityFund("FDCPX",  "Fidelity Select Tech Hardware",        "XLK",  true,  30),
    FidelityFund("FIDSX",  "Fidelity Select Financial Services",   "XLF",  true,  30),
    FidelityFund("FSRBX",  "Fidelity Select Banking",              "XLF",  true,  30),
    FidelityFund("FSPHX",  "Fidelity Select Health Care",          "XLV",  true,  30),
    FidelityFund("FIDRX",  "Fidelity Select Industrials",          "XLI",  true,  30),
    FidelityFund("FSDAX",  "Fidelity Select Defense & Aerospace",  "XLI",  true,  30),
    FidelityFund("FSUTX",  "Fidelity Select Utilities",            "XLU",  true,  30),
    FidelityFund("FSENX",  "Fidelity Select Energy",               "XLE",  true,  30),
    FidelityFund("FSLEX",  "Fidelity Environment & Alt Energy",    "XLU",  true,  30),
    FidelityFund("FRESX",  "Fidelity Real Estate Fund",            "XLRE", true,  30),
    FidelityFund("FSDPX",  "Fidelity Select Materials",            "XLB",  true,  30),
    FidelityFund("FSCSX",  "Fidelity Select IT Services",          "XLK",  true,  30),
    FidelityFund("FBGRX",  "Fidelity Blue Chip Growth",            "QQQ",  false, 0),
)

fun getFundByTicker(ticker: String): FidelityFund? = FIDELITY_FUNDS.find { it.ticker == ticker }

fun getFundsByEtf(etf: String): List<FidelityFund> = FIDELITY_FUNDS.filter { it.parentEtf == etf }
